pkg/providers/tinkerbell/template.go

```diff
@@ -614,3 +622,25 @@ func generateTemplateBuilder(clusterSpec *cluster.Spec) (providers.TemplateBuild
 		time.Now)
 	return templateBuilder, nil
 }
+
+func fillProxyConfigurations(values map[string]interface{}, clusterSpec *cluster.Spec, datacenterSpec v1alpha1.TinkerbellDatacenterConfigSpec) {
```

Comment:
Can you modify this function to return the values map instead of modifying it in-place?
something like 
```
func fillProxyConfigurations(clusterSpec *cluster.Spec, datacenterSpec v1alpha1.TinkerbellDatacenterConfigSpec) map[string]interface{}
```